bids/visualize.py

```diff
@@ -0,0 +1,758 @@
+from __future__ import annotations
+
+import math
+import warnings
+from pathlib import Path
+import re
+from typing import Any
+
+from bids import BIDSLayout
+import numpy as np
+import pandas as pd
+import plotly.express as px
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+
+class EventPlotter:
+    """Class to handle plotting the content of events files.
+
+    Parameters
+    ----------
+    events_file : str | Path
+        Path to the events file.
+
+    event_column : str | None, optional
+        Name of the column containing the events.
+        If `None`, events types are assumed to be in the "trial_type" column.
+
+    include : list[str] | None, optional
+        List of events type to include in the plot.
+    """
+
+    def __init__(
+        self,
+        events_file: str | Path,
+        event_column: str | None = None,
+        include: list[str] | None = None,
+    ):
+        self.COLOR_LIST = px.colors.qualitative.Plotly
+        self.FONT_SIZE = 14
+        self.STANDOFF = 16
+        self.LINE_WIDTH = 3
+        self.GRID_COLOR = "black"
+        self.AXES_COLOR = "black"
+        self.AXES_LINE_WIDTH = 2
+        self.BG_COLOR = "rgb(255,255,255)"
+        self.EVENT_HEIGHT = 1
+        self.FAST_RESPONSE_THRESHOLD = 0.2
+        self.NB_BINS = 40
+        self.TWO_COLUMN_WIDTHS = [0.7, 0.3]
+        self.THREE_COLUMN_WIDTHS = [0.7, 0.15, 0.15]
+        self.FOUR_COLUMN_WIDTHS = [0.7, 0.1, 0.1, 0.1]
+        self.TICK_LENGTH = 6
```

Comment:
If these are constants, they could go in the class definition instead of `__init__`:

```Python
class EventPlotter:
    COLOR_LIST = px.colors.qualitative.Plotly
```

They'll still be manually overridable on individual `EventPlotter` objects with `EventPlotter().TICK_LENGTH = 4` without modifying them for the whole class, but constants that can't be set through `__init__()` have more the feel of class variables than instance variables.